addons/account/models/account_invoice.py

```diff
@@ -1754,12 +1754,17 @@ def _onchange_uom_id(self):
         result = {}
         if not self.uom_id:
             self.price_unit = 0.0
-        else:
+        elif self.uom_id and self.product_id:
```

Comment:
Actually, why is there an `else` here? We have a `if self.product_id and self.uom_id:` below where we can correctly compute the prices. Would it not work if we move the code there? This way we don't need to display an annoying warning, we just don't do anything if somebody select a uom without a product (it will be checked when selecting the product anyway)

Reply:
"Who merged that crap? Oh never mind..." cf #18719

Reply:
You're right, updated